be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/WitnessHandler.scala

```diff
@@ -34,24 +35,21 @@ case object WitnessHandler extends MessageHandler {
     val decodedData: WitnessMessage = rpcMessage.getDecodedData.get.asInstanceOf[WitnessMessage]
     val messageId: Hash = decodedData.message_id
 
-    // val ask = dbActor.ask(ref => DbActor.Read(rpcMessage.getParamsChannel, modificationId, ref)).map {
-    val ask = dbActor.ask("TODO").map {
-      case Some(message: Message) =>
-        /*
-        val askWrite = dbActor.ask(ref => DbActor.Write(
-          rpcMessage.getParamsChannel,
-          message.addWitnessSignature(WitnessSignaturePair(rpcMessage.getParamsMessage.get.sender, decodedData.signature)),
-          ref
-        )).map { */
-        val askWrite = dbActor.ask("m").map {
-          case true =>
-            // FIXME propagate
-            Left(rpcMessage)
-          case _ => Right(PipelineError(-10, "", rpcMessage.id)) // FIXME add DbActor "answers" with error description if failed
-        }
-        Await.result(askWrite, DbActor.getDuration)
-      case None => Right(PipelineError(-10, "", rpcMessage.id)) // FIXME add DbActor "answers" with error description if failed
+    // get message (Message) with messageId message_id from db
+    var message: Message = ???
+
+    // add new witness signature to existing ones
+    message = message.addWitnessSignature(WitnessSignaturePair(message.sender, message.signature))
+
+    // overwrite message in db
+    val f: Future[GraphMessage] = (dbActor ? DbActor.Write(rpcMessage.getParamsChannel, message)).map {
+      case DbActorWriteAck =>
+        // TODO propagate
```

Comment:
Ehm? :)

Reply:
Added it to the TODO list which shortens as we speak ^^. I will now merge this PR so that future PRs mergeable are not stuck